CRM/Core/Payment/PayPalIPN.php

```diff
@@ -394,6 +394,8 @@ public function getInput(&$input, &$ids) {
     $paymentDate = $this->retrieve('payment_date', 'String', FALSE);
     if (!empty($paymentDate)) {
       $receiveDateTime = new DateTime($paymentDate);
+      $systemTimeZone = new DateTimeZone(date_default_timezone_get());
```

Comment:
I think it should be changed to the following so that we only ever have one function calling `date_default_timezone_get()`
```suggestion
      $systemTimeZone = new DateTimeZone(CRM_Core_Config::singleton()->userSystem->getTimeZoneString());
```

Reply:
Yeah, I was wondering about that (I knew where that method lived, but apparently I would have gotten it the wrong way).

I decided to go with `date_default_timezone_get()` initially since that's basically all `getTimeZoneString()` is calling.

Reply:
@jgillmanjr there is actually a material difference in Drupal, etc They overwrite the function as per https://github.com/civicrm/civicrm-core/blob/master/CRM/Utils/System/Drupal.php#L787 which allows them to get the tz of the logged in user first if not fall through various drupal stuff but then back to `date_default_timezone_get` just FYI

Reply:
Ahh! Ok, that's good to know.

I was looking at the definition in `CRM/Utils/System/Base.php`